src/aind_metadata_mapper/gather_metadata.py

```diff
@@ -62,72 +74,376 @@ def _get_file_from_user_defined_directory(self, file_name: str) -> dict:
+                    "instrument": core_metadata.get("instrument"),
+                    "processing": core_metadata.get("processing"),
+                    "quality_control": core_metadata.get("quality_control"),
+                    "model": core_metadata.get("model"),
+                },
+                name=name,
+                location="",  # Location will be set by the transfer service
+            )
+            return metadata
+
     def run_job(self) -> None:
         """Run job"""
         logging.info("Starting run_job")
-        core_metadata = dict()
-        core_metadata[Subject.default_filename()] = self.get_subject()
-        core_metadata[Procedures.default_filename()] = self.get_procedures()
-
-        for k, v in core_metadata.items():
-            logging.debug(f"Writing {k} file")
-            output_file = os.path.join(self.settings.metadata_dir, k)
-            with open(output_file, "w") as f:
-                json.dump(v, f, indent=3, ensure_ascii=False, sort_keys=True)
+
+        # Gather all core metadata
+        core_metadata = {}
+
+        # Always create data description (required)
+        data_description = self.build_data_description()
+        if data_description:
+            core_metadata["data_description"] = data_description
+            self._write_json_file(DataDescription.default_filename(), data_description)
+
+        # Get other metadata (optional)
+        subject = self.get_subject()
+        if subject:
+            core_metadata["subject"] = subject
+            self._write_json_file(Subject.default_filename(), subject)
+
+        procedures = self.get_procedures()
+        if procedures:
+            core_metadata["procedures"] = procedures
+            self._write_json_file(Procedures.default_filename(), procedures)
+
+        acquisition = self.get_acquisition()
+        if acquisition:
+            core_metadata["acquisition"] = acquisition
+            self._write_json_file(Acquisition.default_filename(), acquisition)
+
+        instrument = self.get_instrument()
+        if instrument:
+            core_metadata["instrument"] = instrument
+            self._write_json_file(Instrument.default_filename(), instrument)
+
+        processing = self.get_processing()
+        if processing:
+            core_metadata["processing"] = processing
+            self._write_json_file(Processing.default_filename(), processing)
+
+        quality_control = self.get_quality_control()
+        if quality_control:
+            core_metadata["quality_control"] = quality_control
+            self._write_json_file(QualityControl.default_filename(), quality_control)
+
+        model = self.get_model()
+        if model:
+            core_metadata["model"] = model
+            self._write_json_file(Model.default_filename(), model)
+
+        # Validate and create full metadata object
+        metadata = self.validate_and_create_metadata(core_metadata)
+
+        if isinstance(metadata, dict):
```

Comment:
@helen-m-lin recently pushed a registration API, so the metadata.nd.json file no longer needs to be created to trigger events. We can probably remove creating the metadata.nd.json file.